lib/srv/discovery/status.go

```diff
@@ -293,5 +300,232 @@ func (s *Server) ReportEC2SSMInstallationResult(ctx context.Context, result *ser
 
 	s.updateDiscoveryConfigStatus(result.DiscoveryConfig)
 
+	s.awsEC2Tasks.addFailedEnrollment(
+		awsEC2FailedEnrollmentTaskKey{
+			integration: result.IntegrationName,
+			issueType:   result.IssueType,
+			accountID:   result.SSMRunEvent.AccountID,
+			region:      result.SSMRunEvent.Region,
+		},
+		&usertasksv1.DiscoverEC2Instance{
+			// TODO(marco): add instance name
+			InvocationUrl:   result.SSMRunEvent.InvocationURL,
+			DiscoveryConfig: result.DiscoveryConfig,
+			DiscoveryGroup:  s.DiscoveryGroup,
+			SyncTime:        timestamppb.New(result.SSMRunEvent.Time),
+			InstanceId:      result.SSMRunEvent.InstanceID,
+		},
+	)
+
 	return nil
 }
+
+// awsEC2Tasks contains the Discover EC2 User Tasks that must be reported to the user.
+type awsEC2Tasks struct {
+	mu sync.RWMutex
+	// instancesIssues maps the Discover EC2 User Task grouping parts to a set of instances metadata.
+	instancesIssues map[awsEC2FailedEnrollmentTaskKey]map[string]*usertasksv1.DiscoverEC2Instance
+	// groupsToSync is used to register which groups were changed in memory but were not yet sent to the cluster.
+	// When upserting User Tasks, if the group is not in groupsToSync,
+	// then the cluster already has the latest version of this particular group.
+	taskKeysToSync map[awsEC2FailedEnrollmentTaskKey]struct{}
+}
+
+// awsEC2FailedEnrollmentTaskKey identifies a UserTask group.
+type awsEC2FailedEnrollmentTaskKey struct {
```

Comment:
Nit: Can we shorten the name?
```suggestion
// awsEC2TaskKey identifies a UserTask group.
type awsEC2TaskKey struct {
```